optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,474 @@
+
+                if "best_configure" in state_dict and state_dict["best_configure"] is not None:
+                    try:
+                        model = load(state_dict_path, model)
+                    except Exception as e:
+                        if msg is not None:
+                            e.args += (msg,)
+                        raise
+
+            model.eval()
+
+        return cls(
+            model,
+            config=config,
+            model_save_dir=model_save_dir,
+            use_cache=use_cache,
+            **kwargs,
+        )
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        use_cache: bool = True,
+        torch_dtype: Optional[Union[str, "torch.dtype"]] = None,
+        **kwargs,
+    ):
+        """
+        Export a vanilla Transformers model into a TorchScript model using `torch.jit.trace`.
+
+        Arguments:
+            model_id (`str` or `Path`):
+                The directory from which to load the model.
+                Can be either:
+                    - The model id of a pretrained model hosted inside a model repo on huggingface.co.
+                    - The path to a directory containing the model weights.            save_dir (`str` or `Path`):
+                The directory where the exported ONNX model should be saved, default to
+                `transformers.file_utils.default_cache_path`, which is the cache directory for transformers.
+            config (`PretrainedConfig`) :
+                an object of PretrainedConfig.
+            use_auth_token (`str` or `bool`):
+                Is needed to load models from a private repository
+            revision (`str`):
+                Revision is the specific model version to use. It can be a branch name, a tag name, or a commit id
+            kwargs (`Dict`, *optional*):
+                kwargs will be passed to the model during initialization
+        """
+        if is_torch_version("<", "2.0.0"):
+            raise ImportError("`torch>=2.0.0` is needed to trace your model")
+
+        task = cls.export_feature
+        model_kwargs = {
+            "revision": revision,
+            "use_auth_token": use_auth_token,
+            "cache_dir": cache_dir,
+            "subfolder": subfolder,
+            "local_files_only": local_files_only,
+            "force_download": force_download,
+            "torch_dtype": torch_dtype,
+        }
+
+        if config.torch_dtype != "int8" and config.torch_dtype != torch.int8:
+            model = TasksManager.get_model_from_task(task, model_id, **model_kwargs)
+        else:
+            file_name = kwargs.get("file_name", WEIGHTS_NAME)
```

Comment:
I think it depends on the save_pretrained function, in the save_pretrained function, we allow a user to specify file name, then we need to allow a user to load the specific model name. Do you think so?

Reply:
I would be in favor of removing it as well yes ! what do you think ?

Reply:
But in _save_pretrained function, it allow the user to specify file name, how do we handle the specifical file which user used.(for example: user quantized a model and save it to "int8_model.bin".

Reply:
What about adding a warning stating that this will be deprecated in the next release ? Also adding a warning when this parameter is being disregarded (when loading a model which is not quantized)

Reply:
Added warning message and remove the file_name in the save_pretrained function.